src/irmin-pack/unix/dispatcher.ml

```diff
@@ -225,23 +225,57 @@ module Make (Fm : File_manager.S with module Io = Io.Unix) :
     | Prefix -> Io.read_exn (get_prefix t) ~off:poff ~len buf
     | Suffix -> Suffix.read_exn (Fm.suffix t.fm) ~off:poff ~len buf
 
-  let read_in_prefix_and_suffix_exn t ~off ~len buf =
-    let ( -- ) a b = a - b in
+  let read_bytes_exn t ~f ~off ~len =
```

Comment:
That function is not tied to the use case of transferring data. It can be used to read and decode data. It's not the recommended way to go but it's possible.